datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActions.java

```diff
@@ -77,21 +119,29 @@ protected SdxContext createFlowContext(FlowParameters flowParameters, StateConte
             @Override
             protected void prepareExecution(DatalakeDatabaseRestoreStartEvent payload, Map<Object, Object> variables) {
                 super.prepareExecution(payload, variables);
-                variables.put(BACKUP_ID, payload.getBackupId());
+                if (!variables.containsKey(BACKUP_ID)) {
+                    variables.put(BACKUP_ID, payload.getBackupId());
+                }
                 if (!Strings.isNullOrEmpty(payload.getRestoreId())) {
                     variables.put(RESTORE_ID, payload.getRestoreId());
                 }
-                variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+                if (!variables.containsKey(OPERATION_ID)) {
+                    variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+                }
             }
 
             @Override
             protected void doExecute(SdxContext context, DatalakeDatabaseRestoreStartEvent payload, Map<Object, Object> variables) {
-                LOGGER.info("Datalake database restore has been started for {}", payload.getResourceId());
-                sdxBackupRestoreService.databaseRestore(payload.getDrStatus(),
-                        payload.getResourceId(),
-                        payload.getBackupId(),
-                        payload.getBackupLocation());
-                sendEvent(context, DATALAKE_DATABASE_RESTORE_IN_PROGRESS_EVENT.event(), payload);
+                if (payload.getBackupLocation() != null) {
+                    LOGGER.info("Datalake database restore has been started for {}", payload.getResourceId());
+                    sdxBackupRestoreService.databaseRestore(payload.getDrStatus(),
+                            payload.getResourceId(),
+                            payload.getBackupId(),
+                            payload.getBackupLocation());
+                    sendEvent(context, DATALAKE_DATABASE_RESTORE_IN_PROGRESS_EVENT.event(), payload);
+                } else {
```

Comment:
If the backup location is not provided, FLow will be stuck in DATALAKE_DATABASE_RESTORE_START_STATE. 
Please change login to throw an exception when the backup location is not provided. This will result in DatalakeDatabaseRestoreCouldNotStartEvent

Reply:
made the change below. throw new BadRequestException

Reply:
Normally, we would do such quick semantic validations before the flow has started.

Reply:
@pdarvasi 


Thanks, Now I added @NotEmpty validation at the sdxendpoint.java so that not empty backup location is validated. 

if it is empty, it will return below error before starting the flow. 
{
    "validationErrors": {
        "restoreDatalakeByName.backupLocation": "must not be empty",
        "restoreDatalakeByName.arg2": "must not be empty"
    }
}

Reply:
removed  backup location parameter in endpoint. no longer apply.